backend/logic/users/users.js

```diff
@@ -67,7 +74,77 @@ var getCompletedCourses = async studentID => {
   return courses;
 };
 
+var login = async () => {
+	
+	let LocalStrategy = require('passport-local').Strategy;
+		
+	//LOCAL SIGNIN
+	app.passport.use('local-signin', new LocalStrategy(  
+	{
+		// by default, local strategy uses username and password, we will override with email
+		usernameField : 'username',
+		passwordField : 'password',
+		passReqToCallback : true // allows us to pass back the entire request to the callback
+	},
+	
+	function(req, username, password, done) {
+		let error;
+
+		let isValidPassword = function(userpass, password){
+			return bCrypt.compareSync(password, userpass);
```

Comment:
Just curious, How does bceypt compare without a salt to rehash agaisnt?  Like when you hashed tbe password, you needed to hand in a salt to create the hash, doesnt it need that same salt in order to decrypt the password in the database?